src/App.tsx

```diff
@@ -1,27 +1,71 @@
-import React from 'react';
-import './App.scss';
+import React, { useState } from 'react';
+
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+
+import { getAll, get5First, getRedGoods } from './api/goods';
+
+import './App.scss';
+
+enum GoodsType {
+  Default,
+  All,
+  FirstFive,
+  OnlyRed,
+}
+
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [currentGoodsType, setCurrentGoodsType] = useState(GoodsType.Default);
+  const [isLoadingError, setIsLoadingError] = useState(false);
+
+  const handleLoad = (
+    getGoods: () => Promise<Good[]>,
+    goodsType: GoodsType = GoodsType.Default,
+  ) => {
+    return () => {
+      if (currentGoodsType !== goodsType) {
+        getGoods()
+          .then(setGoods)
+          .catch(() => setIsLoadingError(true));
+        setCurrentGoodsType(goodsType);
+      }
```

Comment:
But what if user have slow internet connection? Let's add loader, now it looks buggy

https://user-images.githubusercontent.com/84275496/226944731-03163102-ad0d-45a3-9ea5-81d8d7d52140.mov